council/utils/code_parser.py

```diff
@@ -44,13 +45,28 @@ def find_last(language: Optional[str] = None, text: str = "") -> Optional[CodeBl
         blocks = CodeParser._build_generator(language, text)
         return last(blocks, None)
 
-    @staticmethod
-    def _get_pattern(language: Optional[str]):
-        return r"```([^\n]*)\n(.*?)\n?```" if language is None else rf"```({re.escape(language)})\n(.*?)\n?```"
-
     @staticmethod
     def _build_generator(language: Optional[str], text: str = "") -> Iterable[CodeBlock]:
-        p = CodeParser._get_pattern(language)
-        matches = re.finditer(p, text, re.DOTALL)
-        for match in matches:
-            yield CodeBlock(match.group(1), match.group(2))
+        lines = text.split("\n")
+        i = 0
+
+        while i < len(lines):
```

Comment:
the for loop was better IMO as it avoid messing with the index variable `i`, and avoid array indexing